src/Std/Data/DTreeMap/Internal/Lemmas.lean

```diff
@@ -1465,4 +1466,109 @@ theorem getThenInsertIfNew?!_snd [TransOrd α] (h : t.WF) {k : α} {v : β} :
 
 end Const
 
+theorem length_keys [TransOrd α] (h : t.WF) :
+    t.keys.length = t.size := by
+  simp_to_model using List.length_keys_eq_length
+
+theorem isEmpty_keys :
+    t.keys.isEmpty = t.isEmpty  := by
+  simp_to_model using List.isEmpty_keys_eq_isEmpty
+
+theorem contains_keys [instBEq : BEq α] [LawfulBEq α] [LawfulBEqOrd α] [TransOrd α] {k : α} (h : t.WF) :
+    t.keys.contains k = t.contains k := by
```

Comment:
I see that the typeclasses are tricky here and I think you have found a very nice solution! Is there any chance of dropping the `LawfulBEq` assumption? As far as  I can tell, the lemmas you invoke should only need `EquivBEq`, and this in turn should follow for both relevant `BEq` instances from `LawfulBEqOrd` and `TransOrd`.